# Appending filtered TRI datasets from years 2007 through 2016, which have been filtered for benzene emissions from the petroleum sector

In [24]:
import pandas as pd

In [25]:
Ben_Petro_2007 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2007.csv")

In [26]:
Ben_Petro_2008 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2008.csv")

In [27]:
Ben_Petro_2009 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2009.csv")

In [28]:
Ben_Petro_2010 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2010.csv")

In [29]:
Ben_Petro_2011 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2011.csv")

In [30]:
Ben_Petro_2012 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2012.csv")

In [31]:
Ben_Petro_2013 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2013.csv")

In [32]:
Ben_Petro_2014 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2014.csv")

In [33]:
Ben_Petro_2015 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2015.csv")

In [34]:
Ben_Petro_2016 = pd.read_csv("C:/Users/Luckbug/Code/emissions/Ben_Petro_2016.csv")

### Add all the datasets together using concat

In [35]:
all = [Ben_Petro_2007, Ben_Petro_2008, Ben_Petro_2009, Ben_Petro_2010, Ben_Petro_2011, Ben_Petro_2012, Ben_Petro_2013, Ben_Petro_2014, Ben_Petro_2015, Ben_Petro_2016]

In [41]:
Ben_10 = pd.concat(all)

In [42]:
Ben_10.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1605 entries, 0 to 166
Columns: 112 entries, Unnamed: 0 to SYS_STSWGMX8O#197_NPUKVK#S5$QM
dtypes: float64(84), int64(10), object(18)
memory usage: 1.3+ MB


In [43]:
Ben_10.head()

Unnamed: 0  YEAR  TRI_FACILITY_ID        FRS_ID  \
0        1216  2007  59101XXNBL700EX  1.100000e+11   
1        2206  2007  77640TXCRFNORTH  1.100000e+11   
2        2483  2007  08861CHVRN1200S  1.100590e+11   
3        2699  2007  79086DMNDSSTARR  1.100000e+11   
4        3100  2007  90222DMNNK2000N  1.100000e+11   

                            FACILITY_NAME             STREET_ADDRESS  \
0            EXXONMOBIL BILLINGS REFINERY          700 EXXONMOBIL RD   
1             MOTIVA-PORT ARTHUR REFINERY          2555 SAVANNAH AVE   
2            BUCKEYE PERTH AMBOY TERMINAL            380 MAURER ROAD   
3         DIAMOND SHAMROCK REFINING CO LP                6701 FM 119   
4  DEMENNOKERDOON DBA WORLD OIL RECYCLING  2000 NORTH ALAMEDA STREET   

          CITY       COUNTY  ST    ZIP               ...                \
0     BILLINGS  YELLOWSTONE  MT  59101               ...                 
1  PORT ARTHUR    JEFFERSON  TX  77640               ...                 
2  PERTH AMBOY    MIDDLESEX  NJ   8861               ...                 
3       SUNRAY        MOORE  TX  79086               ...                 
4      COMPTON  LOS ANGELES  CA  90222               ...                 

   8.6_TREATMENT_ON-SITE 8.7_TREATMENT_OFF-SITE  PROD._WASTE_(8.1_THRU_8.7)  \
0                8300.00                  17.00                   12453.000   
1              131138.00                  13.00                  149075.000   
2                 130.00                   0.00                    2933.000   
3                   0.00                 777.00                   35733.000   
4                2621.08                  20.61                   65130.907   

   8.8_ONE-TIME_RELEASE PROD_RATIO_OR_ACTIVITY  8.9_PRODUCTION_RATIO  \
0                   6.0                    NaN                  0.97   
1                   NaN                    NaN                  0.98   
2                   2.0                    NaN                  0.73   
3                   NaN                    NaN                  1.33   
4                   NaN                    NaN                  0.91   

      PARENT_COMPANY_NAME  PARENT_COMPANY_DB_NUMBER  \
0        EXXON MOBIL CORP                 1213214.0   
1  MOTIVA ENTERPRISES LLC                       NaN   
2   BUCKEYE PARTNERS L.P.               112569892.0   
3      VALERO ENERGY CORP                82686916.0   
4          WORLD OIL CORP                72937436.0   

   SYS_STSITSSXSOLCHJPSAF7486K4RV  SYS_STSWGMX8O#197_NPUKVK#S5$QM  
0                    1.376700e+19                    7.389740e+17  
1                    4.277020e+18                    2.841980e+18  
2                    1.434240e+18                    1.566000e+19  
3                    4.277020e+18                    2.361870e+18  
4                    5.097020e+18                    9.781570e+17  

[5 rows x 112 columns]

### Now I would like to add fugitive emissions and stack emissions together to get total air emissions in a new column.

In [44]:
Ben_10['total_air'] = Ben_10['5.1_FUGITIVE_AIR'] + Ben_10['5.2_STACK_AIR']  # assigned to a column

In [46]:
Ben_10.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1605 entries, 0 to 166
Columns: 113 entries, Unnamed: 0 to total_air
dtypes: float64(85), int64(10), object(18)
memory usage: 1.3+ MB


### Now there are 113 columns, so that's a good sign

In [55]:
Ben_10.total_air.head(5)

0     3970.0
1    14710.0
2     1897.0
3     4346.0
4       10.0
Name: total_air, dtype: float64

In [56]:
Ben_10.to_csv("C:/Users/Luckbug/Code/emissions/Ben_10.csv")